test/torch/test_hook.py

```diff
@@ -293,23 +293,125 @@ def initHidden(self):
+    assert total_norm_remote == total_norm_local
+
+
+# Input: None
+# Output: a local PyTorch tensor with .grad attribute set to [4.5, 4.5, 4.5, 4.5]
+# These operations were taken from this tutorial:
+# https://pytorch.org/tutorials/beginner/blitz/autograd_tutorial.html
+def produce_tensor_with_grad():
+    x = torch.ones(2, 2, requires_grad=True)
+    a = torch.randn(2, 2)
+    a = (a * 3) / (a - 1)
+    a.requires_grad_(True)
+    b = (a * a).sum()
+    y = x + 2
+    z = y * y * 3
+    out = z.mean()
+    # backward function is used for computing the gradient of the tensor 'x'
+    out.backward()
+
+    return x
+
+
 def test_remote_gradient_clipping(workers):
-    # Vanishing gradient test
+    """ Vanishing gradient test over
+        gradients of a remote tensor
+    """
     alice = workers["alice"]
-    vanishing_tensor_test = torch.Tensor([-9.8367e23])
-    remote_vanishing_tensor = vanishing_tensor_test.send(alice)
-    vanishing_remote_tensor_clipped = torch.nn.utils.clip_grad(remote_vanishing_tensor, 2)
-    # Has the remote gradient indeed increased?
-    greater_tensor_check = (vanishing_remote_tensor_clipped > remote_vanishing_tensor).copy().get()
+    local_tensor = produce_tensor_with_grad()
+    remote_tensor = local_tensor.send(alice)
+    # initially, remote_tensor.grad is [4.5, 4.5, 4.5, 4.5]
+    # Method operates in place
+    nn.utils.clip_grad_norm_(remote_tensor, 2)
+    tensor_comparison_grad = torch.Tensor([[4.5, 4.5], [4.5, 4.5]])
+    tensor_comp_grad_remote = tensor_comparison_grad.send(alice)
+    # Has the gradient decreased w.r.t. the initial value (i.e. was it clipped)?
+    smaller_tensor_check_remote = remote_tensor.grad < tensor_comp_grad_remote
+
     one_tensors = torch.ones([1], dtype=torch.uint8)
-    assert torch.eq(greater_tensor_check, one_tensors)
+    one_tensors_remote = one_tensors.send(alice)
+
+    assert torch.eq(one_tensors_remote, smaller_tensor_check_remote.all())
 
 
 def test_local_gradient_clipping():
-    # Vanishing gradient test
-    vanishing_tensor_test = torch.Tensor([-9.8367e23])
-    vanishing_tensor_clipped = torch.nn.utils.clip_grad(vanishing_tensor_test, 2)
-    # Has the local gradient indeed increased?
-    greater_tensor_check = vanishing_tensor_clipped > vanishing_tensor_test
+    """ Vanishing gradient test over
+        gradients of a local tensor
+    """
+    local_tensor = produce_tensor_with_grad()
+    # initially, local_tensor.grad is [4.5, 4.5, 4.5, 4.5]
+    # Method operates in place
+    nn.utils.clip_grad_norm_(local_tensor, 2)
+    tensor_comparison_grad = torch.Tensor([[4.5, 4.5], [4.5, 4.5]])
+    # Has the gradient decreased w.r.t. the initial value (i.e. was it clipped)?
+    smaller_tensor_check = local_tensor.grad < tensor_comparison_grad
+
     one_tensors = torch.ones([1], dtype=torch.uint8)
-    assert torch.eq(greater_tensor_check, one_tensors)
+
+    assert torch.eq(one_tensors, smaller_tensor_check.all())
```

Comment:
You don't need the torch.eq or ones tensor. .all returns a boolean (not a tensor)

Reply:
Well, it looks to me like it actually returns a tensor containing a scalar being 0 for False and 1 for True.

![image](https://user-images.githubusercontent.com/4907418/63527563-a2e3fb80-c501-11e9-93bf-1f7201858a5f.png)

So I just changed the check into an equality among integers.